sympy/polys/matrices/tests/test_xxm.py

```diff
@@ -997,3 +1005,87 @@ def test_XXM_qr_empty_matrix_0x2(DM):
     assert R.is_upper
     assert Q.shape == (0, 0)
     assert R.shape == (0, 2)
+
+
+@pytest.mark.parametrize('DM', DMQ_all)
+def test_xxm_fflu_square_matrix(DM):
+    A = DM([[4, 3], [6, 3]])
+    P, L, D, U = A.fflu()
+    assert P.matmul(A) == L.matmul(D.inv()).matmul(U)
```

Comment:
The tests here are all for fields but the whole point of `fflu` is that it works for non-fields (integral domains). The tests should be for `DMZ_all`.